### PR TITLE
prevent paragraph from being ignored in bolded and italics paragraphs

### PR DESCRIPTION
**Pull request by luisbettencourt**
## 🌟 What is the purpose of this PR?

In the hash.dev blog mdx docs the spacing of bold or italic paragraphs is wrong:

image

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1200211978612931/1202405291900008/f) _(internal)_

## 🔍 What does this change?

Our pages weren't rendering paragraphs when they had anything but text inside them, instead just rendering directly what was inside them. ~~Added checks to prevent this from happening when `<strong>` or `<em>` are inside a paragraph.
Maybe the check can be removed entirely, but I'm not sure why it was there in the first place (in BP the check isn't there for paragraphs).~~ Removed the check.

## ❓ How to test this?

1.  Make a whole paragraph bold by wrapping it inside `**` or italics by wrapping it inside `*`;
1.  Check if the paragraph is bold or italicized and if the spacing is correct.

## 📹 Demo

image

### CODE REVIEW

**Comment by nonparibus** [comment]
I don't think this solution is sufficient. For example, if we are to make an entire paragraph a link (as opposed to `strong` or `em`).

**Comment by luisbettencourt** [comment]
Removed the check completely. Checking every page the only difference I noticed was on this link:

From:
image

To:
image

But this seems like the desired behaviour

**Comment by nonparibus** [comment]
Indeed it is. Tagging Teenoh and Ben. Perhaps one of them can speak to what the check was envisaged to be for in the first place.

File: sites/hashdev/src/util/mdxComponents.tsx [hunk 1]
```diff
@@ -22,10 +22,6 @@ export const mdxComponents: Record<string, ComponentType<any>> = {
   Typography,
 
   p: (props: TypographyProps<"p">) => {
-    if (!Array.isArray(props.children) && typeof props.children !== "string") {
-      // eslint-disable-next-line react/jsx-no-useless-fragment -- fragment is required by tsc
-      return <>{props.children}</>;
-    }
```
Comment (benwerner01) [inline sites/hashdev/src/util/mdxComponents.tsx:28]: I'm not really sure what the intentions of this is either, it's not something [we are doing on the BP site](https://github.com/blockprotocol/blockprotocol/blob/main/site/src/components/mdx/mdx-components.tsx#L197-L199). Happy for us to get rid of it, but might be worth getting thoughts from @nathggns (the git blame suggests it was added as part of https://github.com/hashintel/hash/commit/1e19c3327aaaff1c5fd9b18d150b88ecd8fecbc5)

**Comment by nathggns** [comment]
Looking at this now

File: sites/hashdev/src/util/mdxComponents.tsx (hunk 1 above)
Comment (nathggns) [reply sites/hashdev/src/util/mdxComponents.tsx:28]: Yeah I have no recollection of why this is there. It was added in https://github.com/hashintel/hash/commit/076e06dee396f6e845aac3ab5d6bfbbd6b922556 which I wrote, but I have no memory of it. If it works, I've no issue changing this.

**Review by nathggns — APPROVED** [review]

**Pull request merged**